### PR TITLE
Changelog for v2.177.0, changelog tool updates

### PR DESCRIPTION
**Pull request by vEpiphyte**
- Update changelog tool to add model saver, differ, and model awareness options to the output of the format functions.
- Add changelog for v2.177.0 release
- Start changelog tool tests
- Add reference model for v2.176.0

### CODE REVIEW

File: docs/synapse/userguides/model_update_v2_176_0.rst [hunk 1]
```diff
@@ -0,0 +1,27 @@
+.. _userguide_model_v2_176_0:
+
```
Comment (vEpiphyte) [inline docs/synapse/userguides/model_update_v2_176_0.rst:2]: I want to manually cut the old updates out of the 2.x changelog where they are nicely formatted to back fill here.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -6,6 +6,86 @@
...
+  (`#3832 <https://github.com/vertexproject/synapse/pull/3832>`_)
+- Add ``inet:service:message:type`` property to capture message types.
+  (`#3832 <https://github.com/vertexproject/synapse/pull/3832>`_)
+- Added ``meta:rule:type`` taxonomy.
+  (`#3834 <https://github.com/vertexproject/synapse/pull/3834>`_)
+- See :ref:`userguide_model_v2_177_0` for more detailed model changes.
+
+Features and Enhancements
+-------------------------
+- Add ``synapse.lib.auth`` for non-hive based authentication.
```
Comment (invisig0th) [inline CHANGELOG.rst:37]: This one is an example of something that feels both internal and irrelevant to a user

File: CHANGELOG.rst [hunk 3]
```diff
@@ -6,6 +6,86 @@
...
+  (`#3698 <https://github.com/vertexproject/synapse/pull/3698>`_)
+- Add ``synapse.lib.lmdbslab.SafeKeyValu`` class for managing prefix-based
```
Comment (invisig0th) [inline CHANGELOG.rst:39]: kinda this one too :grin:

File: CHANGELOG.rst [hunk 4]
```diff
@@ -6,6 +6,86 @@
...
+- Updated ``view.merge`` command to use ``$view.swapLayer()`` for improved
+  performance.
+  (`#3828 <https://github.com/vertexproject/synapse/pull/3828>`_)
+- Added ``$view.swapLayer()`` API to allow users to start fresh with an
+  existing view.
+  (`#3828 <https://github.com/vertexproject/synapse/pull/3828>`_)
+- Update the ``aiohttp`` library constraints to ``>=3.10.0,<4.0``. Update the
+  ``aiohttp-socks`` library constraints to ``>=0.10.0,<0.11.0``.
+  (`#3830 <https://github.com/vertexproject/synapse/pull/3830>`_)
+- Tightened up ``aha.svc.list`` Storm command output when using --nexus.
```
Comment (Cisphyx) [inline CHANGELOG.rst:75]:
```suggestion
- Tightened up ``aha.svc.list`` Storm command output when using ``--nexus``.
```

**Review by invisig0th — DISMISSED** [review]

File: docs/synapse/userguides/index_model_updates.rst [hunk 5]
```diff
@@ -0,0 +1,13 @@
+.. _userguide_model_updates:
+
+Model Updates
+#############
+
+The following are model updates which have been made with different synapse releases.
```
Comment (Cisphyx) [inline docs/synapse/userguides/index_model_updates.rst:6]:
```suggestion
The following are model updates which have been made with different Synapse releases.
```

**Review by invisig0th — APPROVED** [review]

File: docs/synapse/userguides/model_update_v2_176_0.rst [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+v2.176.0 Model Updates
+######################
+
+The following model updates were made during the ``v2.176.0`` Synapse release.
+
+**New Forms**
```
Comment (Cisphyx) [inline docs/synapse/userguides/model_update_v2_176_0.rst:8]: Should these be in the same format as the v2.177.0 updates or is that something to be cleaned up later?

**Review by invisig0th — APPROVED** [review]

File: docs/synapse/userguides/model_update_v2_176_0.rst (hunk 6 above)
Comment (vEpiphyte) [reply docs/synapse/userguides/model_update_v2_176_0.rst:8]: I want to clean them up later.

**Review by Cisphyx — APPROVED** [review]

**Pull request merged**